galleries/examples/text_labels_and_annotations/label_subplots.py

```diff
@@ -3,54 +3,50 @@
 Labelling subplots
 ==================
 
-Labelling subplots is relatively straightforward, and varies,
-so Matplotlib does not have a general method for doing this.
+Labelling subplots is relatively straightforward, and varies, so Matplotlib
+does not have a general method for doing this.
 
-Simplest is putting the label inside the axes.  Note, here
-we use `.pyplot.subplot_mosaic`, and use the subplot labels
-as keys for the subplots, which is a nice convenience.  However,
-the same method works with `.pyplot.subplots` or keys that are
-different than what you want to label the subplot with.
+The examples here rely on the ability of `~.Axes.annotate` to conveniently
+position text at a given physical offset (in points) away from a corner of
+the Axes.
+
+Note, here we use `.pyplot.subplot_mosaic`, and use the subplot labels as keys
+for the subplots, which is a nice convenience.  However, the same method works
+with `.pyplot.subplots` or keys that are different than what you want to label
+the subplot with.
 """
 
 import matplotlib.pyplot as plt
 
-import matplotlib.transforms as mtransforms
+# %%
+# Put the label at the top left corner (axes fraction (0, 1)), offset 10 pixels
+# right and 5 pixels down (offset points (+10, -5)), i.e. just inside the axes.
 
 fig, axs = plt.subplot_mosaic([['a)', 'c)'], ['b)', 'c)'], ['d)', 'd)']],
                               layout='constrained')
-
 for label, ax in axs.items():
-    # label physical distance in and down:
-    trans = mtransforms.ScaledTranslation(10/72, -5/72, fig.dpi_scale_trans)
-    ax.text(0.0, 1.0, label, transform=ax.transAxes + trans,
-            fontsize='medium', verticalalignment='top', fontfamily='serif',
-            bbox=dict(facecolor='0.7', edgecolor='none', pad=3.0))
-
-plt.show()
+    ax.annotate(
+        label, (0, 1), (+10, -5), 'axes fraction', 'offset points',
```

Comment:
```suggestion
        label, xy=(0, 1), xytext=(+10, -5), xycoords='axes fraction', textcoords='offset points',
```
Having five positional arguments is difficult to understand.

Reply:
Actually I disagree here, because the order (xy1, xy2, trans1, trans2) is clearer without the kwargs (and with the comment above, which I've now moved closer), which are slightly poorly named.  If the kwargs names were e.g. `annotate(label, xyref=..., xytext=..., coordsref=..., coordstext=...)` or `annotate(label, refxy=..., textxy=..., refcoords=..., textcoords=...)` I'd agree with you as the symmetry would be more apparent.

As things are, I guess if we really want to put in kwargs names we should also reorder the parameters, i.e.
```python
ax.annotate(
    label,
    xy=(0, 1), xycoords="axes fraction",
    xytext=(+10, -5), textcoords="offset points",
    ...)
```
which I'm also OK with if that's what you prefer.

Reply:
This makes sense to me.

Reply:
done